### PR TITLE
Add i2c-communication example

### PR DESCRIPTION
**Pull request by fletch3555**
There very well may be a better way to do this.  I'm more of a java dev than a python dev, so please feel free to tear this apart.

Specifically, Java requires a few calls to convert the String to a char[] then to a byte[].  Here, I've simply converted the string to a bytearray.

I'm also using I2C.writeBulk() instead of I2C.transaction() per the docstring on the transaction method.  I can switch it back if you prefer.

### CODE REVIEW

File: i2c-communication/robot.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        """Robot initialization function"""
+
+        self.arduino = wpilib.I2C(self.PORT, self.DEVICE_ADDRESS)
+
+    def writeString(self, input: string):
+        # Creates a char array from the input string
+        chars = bytearray(input, "ascii")
+
+        # Writes bytes over I2C
+        self.arduino.writeBulk(chars)
```
Comment (virtuald) [inline i2c-communication/robot.py:26]: input is a python function, so let's call this variable something else, like... `s`? 

You can do `chars.encode('utf-8')` for this. `bytearray` is really for things that need to be modified.

File: i2c-communication/robot.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        # For example, "RET043" would indicate that the robot is on the red
+        # alliance, enabled in teleop mode, with 43 seconds left in the match.
+        allianceString = "U"
+        alliance = wpilib.DriverStation.getAlliance()
+        if alliance:
+            allianceString = (
+                "R" if alliance == wpilib.DriverStation.Alliance.kRed else "B"
+            )
+
+        stateMessage = "".join(
```
Comment (virtuald) [inline i2c-communication/robot.py:45]: f-strings are all the rage these days:

```python3
enabledString = "E" if wpilib.DriverStation.isEnabled() else "D",
autoString = "A" if wpilib.DriverStation.isAutonomous() else "T"
matchTime = wpilib.DriverStation.getMatchTime()

msg = f"{allianceString}{enabledString}{autoString}{matchTime:03f}"
```

File: i2c-communication/robot.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+class MyRobot(wpilib.TimedRobot):
+    PORT = wpilib.I2C.Port.kOnboard
+    DEVICE_ADDRESS = 4
+
+    def robotInit(self):
+        """Robot initialization function"""
+
+        self.arduino = wpilib.I2C(self.PORT, self.DEVICE_ADDRESS)
+
+    def writeString(self, input: string):
```
Comment (virtuald) [inline i2c-communication/robot.py:21]: `s: str`

**Review by virtuald — CHANGES_REQUESTED** [review]

File: i2c-communication/robot.py (hunk 2 above)
Comment (fletch3555) [reply i2c-communication/robot.py]: oooo.... I figured there was a better way to do this.  Definitely seems more pythonic.  Thanks!

File: i2c-communication/robot.py (hunk 1 above)
Comment (fletch3555) [reply i2c-communication/robot.py:24]: Good to know, thanks!

File: i2c-communication/robot.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        # is sent in format "AEM###" where A is the alliance color, (R)ed or
+        # (B)lue, E is the enabled state, (E)nabled or (D)isabled, M is the
+        # operation mode, (A)utonomous or (T)eleop, and ### is the zero-padded
+        # time remaining in the match.
+        #
+        # For example, "RET043" would indicate that the robot is on the red
+        # alliance, enabled in teleop mode, with 43 seconds left in the match.
+        allianceString = "U"
+        alliance = wpilib.DriverStation.getAlliance()
+        if alliance:
```
Comment (auscompgeek) [inline i2c-communication/robot.py:38]: I think we should be explicit here about what we're checking.
```suggestion
        if alliance is not None:
```

Comment (fletch3555) [reply i2c-communication/robot.py]: Good call, thanks!

**Pull request merged**